.github/workflows/code_test_and_deploy.yml

```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - main
     tags:
-      - '*'
+      - 'v**'
```

Comment:
In this case, we will not have multi-segment tags (we use semantic versioning only), as such 'v*' will suffice here (and the other change).